src/aap_eda/tasks/analytics.py

```diff
@@ -0,0 +1,74 @@
+#  Copyright 2024 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+from datetime import datetime, timezone
+
+import django_rq
+from django.core.management import call_command
+from rq.exceptions import NoSuchJobError
+
+from aap_eda.conf import application_settings
+from aap_eda.core.tasking import Job, unique_enqueue
+
+logger = logging.getLogger(__name__)
+
+
+ANALYTICS_SCHEDULE_JOB_ID = "gather_analytics"
+ANALYTICS_JOB_ID = "job_gather_analytics"
+ANALYTICS_TASKS_QUEUE = "default"
+
+
+def schedule_gather_analytics() -> None:
+    scheduler = django_rq.get_scheduler()
+    func = "aap_eda.tasks.analytics.gather_analytics"
+    scheduled_time = datetime.now(timezone.utc)
+    interval = application_settings.AUTOMATION_ANALYTICS_GATHER_INTERVAL
+    logger.info(
+        f"Adding periodic job {ANALYTICS_SCHEDULE_JOB_ID} "
+        f"to run every {interval} seconds"
+    )
+    scheduler.schedule(
+        scheduled_time=scheduled_time,
+        func=func,
+        interval=interval,
+        id=ANALYTICS_SCHEDULE_JOB_ID,
+    )
+
+
+def reschedule_gather_analytics(new_interval: int, serializer=None) -> None:
+    try:
+        job = Job.fetch(ANALYTICS_SCHEDULE_JOB_ID, serializer=serializer)
+    except NoSuchJobError:
+        logger.warning(f"Job {ANALYTICS_SCHEDULE_JOB_ID} does not exist")
+        return
+    job.meta["interval"] = new_interval
+    job.save()
+    logger.info(
+        f"Reconfigure periodic job {ANALYTICS_SCHEDULE_JOB_ID} with "
+        f"new interval {new_interval} seconds"
+    )
+
+
+def gather_analytics(queue_name: str = ANALYTICS_TASKS_QUEUE) -> None:
+    logger.info("Queue EDA analytics")
+    unique_enqueue(queue_name, ANALYTICS_JOB_ID, _gather_analytics)
+
+
+def _gather_analytics() -> None:
+    if not application_settings.INSIGHTS_TRACKING_STATE:
+        logger.info("INSIGHTS_TRACKING_STATE is not enabled")
+        return
+    logger.info("Collecting EDA analytics")
+    call_command("gather_analytics", "--ship")
```

Comment:
The insights-analytics-collector has `schduled` collection_type, it may be the right type for scheduler tasks. https://github.com/RedHatInsights/insights-analytics-collector/blob/main/insights_analytics_collector/collector.py#L31
In parity with the controller's `gather_analytics` system command, we don't accept `scheduled` options neither.

Reply:
The scheduler cannot set the collection_type unless the system command supports it. Is there a plan for adding the type?

Reply:
The command is parity with the controllers', you can call `AnalyticsCollector` directly, which allow you to set `content_type`